File: app/src/main/java/com/jdappel/beerinvestigator/ui/viewmodel/BeerViewModel.java
package com.jdappel.beerinvestigator.ui.viewmodel;

import com.jdappel.beerinvestigator.rest.Beer;

import java.util.List;

import io.reactivex.Observable;

/**
 * Interface for defining the behavior of the view model.
 */
public interface BeerViewModel {

    void subscribe(Observable<String> searchString, Observable<Boolean> checkbox);

    Observable<List<Beer>> getBeers();

    void unsubscribe();
}

File: app/src/main/java/com/jdappel/beerinvestigator/ui/viewmodel/BeerViewModel.kt
package com.jdappel.beerinvestigator.ui.viewmodel

import androidx.lifecycle.LiveData
import com.jdappel.beerinvestigator.data.model.Brewery
import kotlinx.coroutines.flow.Flow

/**
 * Interface for defining the behavior of the view model.
 */
interface BeerViewModel {
    fun subscribe(searchString: Flow<String>, checkbox: Flow<Boolean>)
    val breweries: LiveData<List<Brewery>>
}